American Whitewater/Helpers/Filters.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  Filters.swift
+//  Filters
+//
+//  Created by Phillip Kast on 7/28/21.
+//  Copyright © 2021 American Whitewater. All rights reserved.
+//
+
+import Foundation
+
+struct Filters: Equatable {
+    // FIXME: currently it appears that the app tries to treat these as a toggle, i.e. if showDistanceFilter = true, showRegionFilter must be false and vice versa. So why have both? Or should it be an enum?
+    // (See below, the predicates mostly
+    var showDistanceFilter: Bool
+    var showRegionFilter: Bool
+    
+    var regionsFilter: [String]
+    var distanceFilter: Double
+    var classFilter: [Int]
+    var runnableFilter: Bool
+    
+    //
+    // MARK: - Some defaults for first run or resetting
+    //
+    
+    static let defaultByRegionFilters = Filters(        
+        showDistanceFilter: false,
+        showRegionFilter: true,
+        regionsFilter: [], // FIXME?
+        distanceFilter: 100,
+        classFilter: [1,2,3,4,5],
+        runnableFilter: false
+    )
+    
+    static let defaultByDistanceFilters = Filters(
+        showDistanceFilter: true,
+        showRegionFilter: false,
+        regionsFilter: [],
+        distanceFilter: 100,
+        classFilter: [1,2,3,4,5],
+        runnableFilter: false
+    )
+}
+
+//
+// MARK: - Predicates
+//
+
+extension Filters {
+    private var runnablePredicate: NSPredicate? {
+        guard runnableFilter else {
```

Comment:
Interesting, not a big deal either way, and I've had this argument before. Even though the cyclomatic complexity is the same, I find if/else to be simpler for two reasons.
1. It's more functional: you can replace and if/else block with another function, and so it naturally modularizes your code
2. The branch logic is more explicit, which makes it easier to reason about what's happening at a particular time. For instance if the function was longer, I would have to keep track of the state of `runnableFilter` and it would be conceivable that another dev could come in and add a redundant check for the same state later in the function. 

That said, lots of people favor the "fail early" pattern, which can definitely be appropriate in some cases. Anyway, I'm fine either way.